### PR TITLE
Feature/autoruncontrol

### PR DESCRIPTION
**Pull request by momoadeli**
Kirill, here is a draft of a working autorun ON/OFF functionality. Please feedback before I move further (cleanup, redundancies, etc)--I want to make sure I leave this in a state that _you_ above all are comfortable with for obvious reasons. BTW, there are no 3d party libs (removed bootstrap). A few quick items:

1.  I see a new src/exercises/codelab-config.ts.  I assume you're transitioning to it (I see better persistence across executions).  As such I've pointed my one file in the new code (namely `autoruncontrol.component.ts`) to it and also added an 'autorun' flag
2.  There are a few ways autorun ON/OFF could be implemented.  Currently I have a 'decoupled' `autoruncontrol.component.ts`) which @Outputs its 'state'  via `OnAutorunChange` to its parent `exercise.component.ts.`. After your feedback, instead of the emitted (@Output), I would like to enhance this where the parent can pass through an `(@Input)` method to this component to use for 'run-click' which I believe is a cleaner pattern.
2A:  Needless to say, the autorunner has a 'toggler'  The proper` [ActionType.TOGGLE_AUTORUN]` method has been added to the `reducers.service.ts`.
3.  `exercise.component.ts`:  Here I've added a `(@Output)OnAutorunChange` which is triggered by `autorun.component.ts`  similar to `(@Output)onCodeChange` which is triggered by its `editors.components.ts` child. Both, one way or another, eventually, execute a `this.state.updateCode(changedFile)` which is handled by the `state.service.ts`. 
4.  I've tried to follow your current styles and patterns with the other knick-knack code I've added to build this here and there.
5.  I'm not a UI designer, so let me know if you'r comfortable with the `autoruncontrol.compoent.ts` look and feel.  There are different ways this can be done too.
6. After I merged your latest, there seems to be new behavior here and there. for example it seems you are disabling user 'looking ahead' in the next tab in the same 'exercise'.
Looking forward to feedback.  Talk soon!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
     "karma-cli": "^1.0.1",
     "karma-jasmine": "^1.0.2",
     "karma-remap-istanbul": "^0.2.1",
-    "protractor": "4.0.9",
+    "protractor": "4.0.11",
```
Comment (kirjs) [inline package.json:47]: This and few files above seem to be coming from a different PR.

File: src/app/action-types.enum.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 export enum ActionTypes {
   LOAD_SOLUTION,
   TOGGLE_FILE,
+  TOGGLE_AUTORUN,//momo
```
Comment (kirjs) [inline src/app/action-types.enum.ts:4]: No need for the comment, it will be in blame history either way

**Comment by kirjs** [comment]
Hey Mo, thanks for looking into this, couple of things before I'll start looking into details: 
1) Please add a screenshot in the description.
2) This branch seems to be based on some other PR, and there are a bunch of protractor changes included from #9, please remove those (let me know if you need help with the git part)
3) No need for named comments, diffs and blame can be used to track the history.

**Comment by momoadeli** [comment]
@kirjs , will do.  BTW, about the protractor items, I updated my fork based on your latest upstream merges.  You seemed to have merged these in.  Should I still discard?
